acvm-repo/blackbox_solver/src/logic.rs

```diff
@@ -8,16 +8,15 @@ pub fn bit_xor<F: AcirField>(lhs: F, rhs: F, num_bits: u32) -> F {
     bitwise_op(lhs, rhs, num_bits, |lhs_byte, rhs_byte| lhs_byte ^ rhs_byte)
 }
 
+/// Performs a bitwise operation on two field elements by treating them as byte arrays.
+///
+/// Both field elements are converted to little-endian byte arrays and masked to keep only
+/// the lowest `num_bits` bits. The provided operation `op` is then applied byte-by-byte,
+/// and the result is converted back to a field element.
+/// This function works for any `num_bits` value and does not assume it to be a multiple of 8.
 fn bitwise_op<F: AcirField>(lhs: F, rhs: F, num_bits: u32, op: fn(u8, u8) -> u8) -> F {
-    // XXX: Gadgets like SHA256 need to have their input be a multiple of 8
-    // This is not a restriction caused by SHA256, as it works on bits
-    // but most backends assume bytes.
-    // We could implicitly pad, however this may not be intuitive for users.
-    // assert!(
-    //     num_bits % 8 == 0,
-    //     "num_bits is not a multiple of 8, it is {}",
-    //     num_bits
-    // );
+    // We could explicitly expect `num_bits` to be a multiple of 8 as most backends assume bytes:
+    // assert!(num_bits % 8 == 0, "num_bits is not a multiple of 8, it is {num_bits}");
```

Comment:
The old comment had the justification for why we don't do this but it was removed for some reason. The new comment leaves me asking "why don't we?"